src/zenml/integrations/github/code_repositories/github_code_repository.py

```diff
@@ -237,19 +237,35 @@ def check_remote_url(self, url: str) -> bool:
         Returns:
             Whether the remote url is correct.
         """
-        host = self.config.host or "github.com"
-        host = host.rstrip("/")
-
-        https_url = (
-            f"https://{host}/{self.config.owner}/{self.config.repository}.git"
-        )
-        if url == https_url:
-            return True
-
-        ssh_regex = re.compile(
-            f".*@{host}:{self.config.owner}/{self.config.repository}.git"
-        )
-        if ssh_regex.fullmatch(url):
-            return True
-
+        # Normalize repository information
```

Comment:
check gitlab integration for a much better way than regex match for the HTTPS URL